pkg/sql/prepared_stmt.go

```diff
@@ -121,6 +121,16 @@ type preparedStatementsAccessor interface {
 // References need to be registered with incRef() and de-registered with
 // decRef().
 type PreparedPortal struct {
+	*immutablePreparedPortal
```

Comment:
What's the reason we need to separate out the types here? Why can't we add exhausted to the existing PreparedPortal type?